dashboard/src/reducers/kube.ts

```diff
@@ -0,0 +1,34 @@
+import { LOCATION_CHANGE, LocationChangeAction } from "connected-react-router";
+import * as _ from "lodash";
+
+import { getType } from "typesafe-actions";
+import actions from "../actions";
+import { KubeAction } from "../actions/kube";
+import { IKubeState } from "../shared/types";
+
+const initialState: IKubeState = {
+  items: {},
+};
+
+const kubeReducer = (
+  state: IKubeState = initialState,
+  action: KubeAction | LocationChangeAction,
+): IKubeState => {
+  switch (action.type) {
+    case getType(actions.kube.requestResource):
+      return { ...state, items: _.merge({}, state.items, action.payload) };
```

Comment:
can't the merge be done by the built-in spread operator instead of using lodash?

Reply:
the `action.payload` is an object with a single key so we would need to do something like:

```
      const key = Object.keys(action.payload)[0];
      return { ...state, items: {...state.items, [key]: action.payload[key] } };
```

Unless there is an easier way I am unaware of it seems for me that it's more readable to use `merge`.

Reply:
I'm not sure I understand, why can't we just do 
```
{ ...state, items: {...state.items, ...action.payload } };
```
?

Reply:
Why doesn't it apply here? And yes, we can discuss it in other issue since it affects every single part of the store.

Reply:
No, I don't mean it doesn't apply, I mean that we would need to properly think how to flatten this and others if necessary.

Reply:
Created https://github.com/kubeapps/kubeapps/issues/865 elaborate your idea in that issue if you need to.